python/hail/dataset.py

```diff
@@ -1034,6 +1034,73 @@ def count(self):
         return r._1(), r._2()
 
     @handle_py4j
+    @typecheck_method(ped=Pedigree,
+                      population_AF=strlike,
+                      min_GQ=integral,
+                      min_p=numeric,
+                      max_parent_AB=numeric,
+                      min_child_AB=numeric,
+                      min_depth_ratio=numeric)
+    def de_novo(self, ped, population_AF, min_GQ=20, min_p=0.05,
+                max_parent_AB=0.05, min_child_AB=0.20, min_depth_ratio=0.10):
+        """Call de novo variation from trio data.
+
+        **Examples**
+
+        Call de novo events and export them to a file:
+
+        >>> pedigree = Pedigree.read('data/myStudy.fam')
+        >>> priors = hc.import_table('data/gnomadFreq.tsv', impute=True).key_by('Variant')
+        >>> denovo_kt = (vds.annotate_variants_table(priors, root='va.gnomAD')
+        ...                  .de_novo(pedigree, 'va.gnomAD'))
+        >>> denovo_kt.export('denovo_calls.tsv')
+
+        **Notes**
+
+        This method replicates the functionality of `Kaitlin Samocha's de novo caller <https://github.com/ksamocha/de_novo_scripts>`_.
+        It is reproduced in Hail with her permission and assistance. See the link above
```

Comment:
If this is going to be Hail's standard vanilla de novo caller, I think it should be fully documented here, not through a link, in the same style as our other documentation. Most of this is copy paste. And I think we should also clearly document the underlying model (for example, I see these priors buried in the code) so users can meaningfully interpret the output. In general, we should hold to the standard of documenting all mathematical models we use (less important to document the algorithms used to fit the models). And if there is a paper corresponding to this method, you should reference that as well.